webapp/src/selectors.js

```diff
@@ -1,9 +1,82 @@
+import {createSelector} from 'reselect';
+
+import {Preferences} from 'mattermost-redux/constants';
+import {get} from 'mattermost-redux/selectors/entities/preferences';
+import {getMyTeams} from 'mattermost-redux/selectors/entities/teams';
+
 import {id as pluginId} from './manifest';
 
+const MESSAGE_DISPLAY = 'message_display';
+const MESSAGE_DISPLAY_DEFAULT = 'clean';
+const MESSAGE_DISPLAY_COMPACT = 'compact';
+
 function getPluginState(state) {
     return state['plugins-' + pluginId] || {};
 }
 
 export function getConfirmationModalState(state) {
     return getPluginState(state).confirmationModal || {};
-}
\ No newline at end of file
+}
+
+export function getWindowWidth(state) {
+    return getPluginState(state).windowWidth;
+}
+
+export function isSurveyPostSmall(state, isRHS) {
+    return isRHS || !useFullSurveyPost(state);
+}
+
+function isCompactView(state) {
+    const compactView = get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, MESSAGE_DISPLAY, MESSAGE_DISPLAY_DEFAULT) === MESSAGE_DISPLAY_COMPACT;
+    return compactView;
+}
+
+function isSidebarOpen(state) {
+    return state.views.rhs.isSidebarOpen;
+}
+
+function isTeamSidebarVisible(state) {
+    return getMyTeams(state).length > 1;
+}
+
+// The required width for the regular and small survey posts with different combinations of display settings
+const requiredWidths = [
```

Comment:
For future consideration, we might want to brainstorm an alternative method for adjusting the UI to fit its container width as hardcoding specific pixel values can be brittle. 🙂

Reply:
Yeah, this isn't great. We could change the RHS to work like the team sidebar where the width is just subtracted from the total width required, but I didn't think of that until after working out these numbers. I'll leave that as something for future improvement